searchlib/src/vespa/searchlib/datastore/bufferstate.cpp

```diff
@@ -80,6 +80,11 @@ calcAllocation(uint32_t bufferId,
     size_t allocClusters = typeHandler.calcClustersToAlloc(bufferId, elementsNeeded, resizing);
     size_t allocElements = allocClusters * typeHandler.getClusterSize();
     size_t allocBytes = roundUpToMatchAllocator(allocElements * typeHandler.elementSize());
+    size_t maxAllocBytes = typeHandler.getMaxClusters() * typeHandler.getClusterSize() * typeHandler.elementSize();
+    if (allocBytes > maxAllocBytes) {
```

Comment:
This will render parts of the memory wasted. I would suggest that we instead reduce the size of the buffer requested to something that we can represent.